popsycle/synthetic.py

```diff
@@ -3512,3 +3527,390 @@ def generate_slurm_scripts(slurm_config_filename, popsycle_config_filename,
+
+    #Defining galactocentric coordinates
+    c = coord.Galactocentric(x=cart[0] * units.kpc, y=cart[1] * units.kpc, z=cart[2] * units.kpc)
+
+    #Transforming from galactocentric to galactic coordinates
+    #outputs l, b, dist in degrees.
+    c = c.transform_to(coord.Galactic(representation_type='cartesian'))
+
+    #Set minimum and maximum l, b, and r values for PBH mask. 
+    #Determining the center of the field of view circle.
+    l_min = np.min(long_bin).values
+    l_max = np.max(long_bin).values
+    b_min = np.min(lat_bin).values
+    b_max = np.max(lat_bin).values
+    l_mid = np.average((l_min, l_max))
+    b_mid = np.average((b_min, b_max))
+
+    dists = angdist(l_mid, b_mid, c.l.deg, c.b.deg)
+    r_max_mask = 16.6
+
+    #Masking the full PBH data for our field of view
+    mask = (c.l.deg >= l_min) & (c.l.deg <= l_max) & (c.b.deg >= b_min) & (c.b.deg <= b_max) & (c.distance.kpc <= r_max_mask) & (dists < field_of_view_area)
+    data_in_field = c[mask]
+
+    #Obtain the radius values for all of the PBHs in our field of view.
+    r = data_in_field.distance.kpc
+
+    #Getting radial dispersion vs. radius distribution from the M. Hoeft 2018 paper.
+    disp_data = pd.read_csv('data/radial_velocity_dispersion_digitized.csv')
+    plot_r_val = disp_data['radius'].values
+    plot_dispersion = disp_data.iloc[:,1].values
+
+    #Adjusting for the correct units
+    plot_r_val_kpc = plot_r_val*(10**3)
+    plot_dispersion_fixed = plot_dispersion*1000
+
+    #Get radial dispersion value for each PBH, given PBH radius.
+    pbh_dispersion = np.interp(r, plot_r_val_kpc, plot_dispersion_fixed)
+
+    #Defining some parameters that are used in the velocity distribution equations
+    v_esc = 525 #km/s
+    v_knot = ((2/3)**(1/2))*pbh_dispersion
+    z=v_esc/v_knot
+    n_esc=scipy.special.erf(z)-(2*(np.pi**(-1/2))*z*(np.exp(-z**2)))
+    v=np.arange(0,525, 525/len(n_esc))
+
+    #Getting velocity for each PBH using formulas from Ranjan Laha 2018.
+    for disp in pbh_dispersion:
+        f = ((3/(2*np.pi*(disp**2)))**(3/2))*(np.exp((-(v**2))/(2*(disp**2))))
+        cdf = (3*(3**(1/2))*(1/(disp**2))*disp*scipy.special.erf(v/((2**(1/2))*disp)))/(4*np.pi)
+        rand_cdf = np.random.uniform(0, np.amax(cdf), 1)
+        interpreted_rms_velocities = np.interp(rand_cdf, cdf, v)
+
+    #Sampling random lattitude and longitude values for velocity to complete the spherical velocities
+    sin_lat_vel = np.random.uniform(-1, 1, len(data_in_field))
+    lat_vel = np.arcsin(sin_lat_vel)
+    long_vel = np.random.uniform(0, 2*np.pi, len(data_in_field))
+
+    #Transforming velocities to cartesian to get vx, vy, vz values
+    cart_vel = astropy.coordinates.spherical_to_cartesian(interpreted_rms_velocities, lat_vel, long_vel)
+    vx= cart_vel[0]
+    vy= cart_vel[1]
+    vz= cart_vel[2]
+
+    #Getting rest of data for PBHs for the combined .h5 file
+    mass = np.full((len(data_in_field), 1), pbh_mass)
```

Comment:
This line, and all subsequent examples of calling `np.full` followed by `np.reshape`, can be changed to `mass = np.full(len(data_in_field), pbh_mass)`